### PR TITLE
ODBC: First version Winsetup

### PR DESCRIPTION
**Pull request by tiagokepe**
Hey all,

This PR provides the first support for the Windows setup .dll and installer.
A new action (Install ODBC) was added into the Windows 64 CI.

### CODE REVIEW

File: tools/odbc/winsetup/install.c [hunk 1]
```diff
@@ -0,0 +1,303 @@
+/*
```
Comment (Mytherin) [inline tools/odbc/winsetup/install.c:1]: Perhaps place a `LICENSE` file in the `winsetup` directory that contains the license information and refers back to the MonetDB repository, e.g. similar to the ones we have [here](https://github.com/duckdb/duckdb/blob/master/third_party/re2/LICENSE).

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. One minor comment:

File: tools/odbc/winsetup/install.c (hunk 1 above)
Comment (tiagokepe) [reply tools/odbc/winsetup/install.c]: Added LICENSE file and adjusted the copyright in the files.

**Comment by alanpaulkwan** [comment]
Is there a walkthrough for how to use this? I'd love to test it.

**Comment by tiagokepe** [comment]
Hi @alanpaulkwan,

The last PR #2550 enabled running some ODBC test cases on Windows. The [odbc-win-64 CI](https://github.com/duckdb/duckdb/pull/2550/files#:~:text=exe-,odbc-win-64%3A,-name) would be a good start point.

Basically, for now, you can enable the ODBC compilation setting the cmake option `-DBUILD_ODBC_DRIVER=1`.

The ODBC artifacts will be created at the folder: `tools/odbc/bin/Release/` (based on the build folder).

Then, you can execute `tools/odbc/bin/Release/odbc_install.exe /Install` to set up all required ODBC Windows registries.

An ODBC DuckDB registry entry will be created with a default database (_:memory:_) at: `HKLM\SOFTWARE\ODBC\ODBC.INI\DuckDB\database` (you can change the database value to point to any file path in your system)

P.S. Some ODBC functionalities are still under development and will be added gradually in future PRs.

**Pull request merged**